src/logger/src/logger.rs

```diff
@@ -645,10 +688,37 @@ mod tests {
         logger.mock_log(Level::Warn, "msg");
         validate_log(
             &mut Box::new(&mut reader),
-            "[TEST-INSTANCE-ID:WARN:logger.rs:0] msg\n",
+            &format!(
+                "[TEST-INSTANCE-ID:{}:WARN:logger.rs:0] msg\n",
+                crnt_thread_name
+            ),
         );
     }
 
+    #[test]
+    fn test_thread_name_custom() {
+        let custom_thread = thread::Builder::new()
+            .name("custom-thread".to_string())
+            .spawn(move || {
+                // test to check the thread name
+                let logger = Logger::mock_new();
+                let mut reader = logger.mock_init();
+                // Test with a mock instance id.
```

Comment:
Is there a special reason we want to set the instance id here? If not, we could remove setting it and let the test verify only if we log the custom name or not, as the test name says :) .

Reply:
When we set up the initial PR, we used an optional parameter to add the thread name after the instance id. We wanted to test the prefix formatting a little more and hence the remnants of that test. You're right! Now that we test the thread name on all the other unit tests, it's better to simplify this one.